Cargo.toml

```diff
@@ -67,17 +67,22 @@ defmt = [
     "stm32h5/defmt",
 ]
 
+adc = ["dep:embedded-hal-02", "dep:nb"]
```

Comment:
We need embedded_hal_02 for the `adc::Channel` trait. However we could probably avoid `nb` if we provide our own blocking `Adc::convert` method for those that do not need to be generic using the `embedded_hal_02::adc::Oneshot` trait.